src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -94,133 +100,153 @@ def time_provider(self):
-                        # If we timeout, that means full_system missed the last
-                        # wrapper and robot status, lets resend it.
-                        logger.warning("Fullsystem missed last wrapper, resending ...")
-
-                        ssl_wrapper = self.ssl_wrapper_buffer.get(block=False)
-                        robot_status = self.robot_status_buffer.get(block=False)
-
-                        self.blue_full_system_proto_unix_io.send_proto(
-                            SSL_WrapperPacket, ssl_wrapper
-                        )
-                        self.blue_full_system_proto_unix_io.send_proto(
-                            RobotStatus, robot_status
-                        )
-
-                # Validate
-                (
-                    eventually_validation_proto_set,
-                    always_validation_proto_set,
-                ) = validation.run_validation_sequence_sets(
-                    world,
-                    eventually_validation_sequence_set,
-                    always_validation_sequence_set,
+                # Set the test name
+                eventually_validation_proto_set.test_name = self.test_name
+                always_validation_proto_set.test_name = self.test_name
+
+                # Send out the validation proto to thunderscope
+                self.thunderscope.blue_full_system_proto_unix_io.send_proto(
+                    ValidationProtoSet, eventually_validation_proto_set
+                )
+                self.thunderscope.blue_full_system_proto_unix_io.send_proto(
+                    ValidationProtoSet, always_validation_proto_set
                 )
 
-                if self.thunderscope:
+            # Check that all always validations are always valid
+            validation.check_validation(always_validation_proto_set)
 
-                    # Set the test name
-                    eventually_validation_proto_set.test_name = self.test_name
-                    always_validation_proto_set.test_name = self.test_name
+        # Check that all eventually validations are eventually valid
+        validation.check_validation(eventually_validation_proto_set)
 
-                    # Send out the validation proto to thunderscope
-                    self.thunderscope.blue_full_system_proto_unix_io.send_proto(
-                        ValidationProtoSet, eventually_validation_proto_set
-                    )
-                    self.thunderscope.blue_full_system_proto_unix_io.send_proto(
-                        ValidationProtoSet, always_validation_proto_set
-                    )
+        self.__stopper()
 
-                # Check that all always validations are always valid
-                validation.check_validation(always_validation_proto_set)
 
-            # Check that all eventually validations are eventually valid
-            validation.check_validation(eventually_validation_proto_set)
+class InvariantTestRunner(SimulatedTestRunner):
```

Comment:
put class docs for what an InvariantTestRunner is

Reply:
done